secret_store/src/key_server_cluster/cluster.rs

```diff
@@ -1263,372 +679,426 @@ pub mod tests {
+		pub fn key_storage(&self, idx: usize) -> &Arc<DummyKeyStorage> {
+			self.key_storages_map.values().nth(idx).unwrap()
+		}
+
+		/// Get keys storage reference by node id.
+		pub fn key_storage_of(&self, node: &NodeId) -> &Arc<DummyKeyStorage> {
+			&self.key_storages_map[node]
+		}
+
+		/// Replace key storage of the node by its id.
+		pub fn replace_key_storage_of(&mut self, node: &NodeId, key_storage: Arc<DummyKeyStorage>) {
+			*self.key_storages_map.get_mut(node).unwrap() = key_storage;
+		}
+
+		/// Get ACL storage reference by its index.
+		pub fn acl_storage(&self, idx: usize) -> &Arc<DummyAclStorage> {
+			self.acl_storages_map.values().nth(idx).unwrap()
+		}
+
+		/// Get sessions container reference by its index.
+		pub fn sessions(&self, idx: usize) -> &Arc<ClusterSessions> {
+			&self.cluster(idx).data.sessions
+		}
+
+		/// Get sessions container reference by node id.
+		pub fn sessions_of(&self, node: &NodeId) -> &Arc<ClusterSessions> {
+			&self.clusters_map[node].data.sessions
+		}
+
+		/// Isolate node from others.
+		pub fn isolate(&self, idx: usize) {
+			let node = self.node(idx);
+			for (i, cluster) in self.clusters_map.values().enumerate() {
+				if i == idx {
+					cluster.data.connections.isolate();
+				} else {
+					cluster.data.connections.disconnect(node.clone());
+				}
+			}
+		}
+
+		/// Exclude node from cluster.
+		pub fn exclude(&mut self, idx: usize) {
+			let node = self.node(idx);
+			for (i, cluster) in self.clusters_map.values().enumerate() {
+				if i != idx {
+					cluster.data.connections.exclude(node.clone());
+				}
+			}
+			self.key_storages_map.remove(&node);
+			self.acl_storages_map.remove(&node);
+			self.key_pairs_map.remove(&node);
+			self.clusters_map.remove(&node);
+		}
+
+		/// Include new node to the cluster.
+		pub fn include(&mut self, node_key_pair: Arc<PlainNodeKeyPair>) -> usize {
+			let key_storage = Arc::new(DummyKeyStorage::default());
+			let acl_storage = Arc::new(DummyAclStorage::default());
+			let cluster_params = ClusterConfiguration {
+				self_key_pair: node_key_pair.clone(),
+				key_server_set: Arc::new(MapKeyServerSet::new(false, self.nodes().iter()
+					.chain(::std::iter::once(node_key_pair.public()))
+					.map(|n| (n.clone(), format!("127.0.0.1:{}", 13).parse().unwrap()))
+					.collect())),
+				key_storage: key_storage.clone(),
+				acl_storage: acl_storage.clone(),
+				admin_public: None,
+				preserve_sessions: self.preserve_sessions,
+			};
+			let cluster = new_test_cluster(self.messages.clone(), cluster_params).unwrap();
+
+			for cluster in self.clusters_map.values(){
+				cluster.data.connections.include(node_key_pair.public().clone());
+			}
+			self.acl_storages_map.insert(node_key_pair.public().clone(), acl_storage);
+			self.key_storages_map.insert(node_key_pair.public().clone(), key_storage);
+			self.clusters_map.insert(node_key_pair.public().clone(), cluster);
+			self.key_pairs_map.insert(node_key_pair.public().clone(), node_key_pair.clone());
```

Comment:
```suggestion
			self.key_pairs_map.insert(*node_key_pair.public(), node_key_pair.clone());
```

Reply:
You could also `deref` the public key before this in order to avoid cloning `node_key_par` and move it instead!